src/process/validation/util.ts

```diff
@@ -92,3 +95,29 @@ export const interpolateErrors = (errors: FieldError[], lang: string = 'en') =>
         };
      });
    };
+
+export const hasValue = (value: any) => {
+    if (_isObject(value)) {
+        return !_isEmpty(value);
+    }
+
+    return (typeof value === 'number' && !Number.isNaN(value)) || !!value;
+}
+
+export const doesArrayDataHaveValue = (dataValue: any[] = []): boolean => {
+  if (!Array.isArray(dataValue)) {
+    return !!dataValue;
+  }
+
+  if (!dataValue.length) {
+    return false;
+  }
+
+  const isArrayDataComponent = dataValue.every(_isPlainObject);
```

Comment:
Would it be possible to use the existing `getModelType` family of functions here?

Reply:
@brendanbond would be problematic to use the mentioned function here, since a couple of upper functions recieve only 'value' from context, but in order to use this func, we need a 'component'. 
Do you think it's worth changing upper functions to recieve component/whole context object?